rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -0,0 +1,260 @@
+# RFC 8216 - 2021-07-19 - multiple pipelines
+
+Large Vector users often require complex Vector topologies to facilitate the collection and processing of data from many different upstream sources. This results in large Vector configuration files that are hard to manage across different teams. This makes Vector a bad candidate for large teams that require autonomy to facilitate sane collaboration.
+
+## Scope
+
+This RFC will cover:
+
+- How and where the pipelines are stored
+- How vector reads the pipelines and what are the limits of pipelines
+- What metadatas vector adds to the events going through a pipeline
+
+This RFC will not cover:
+
+- How the pipelines should be synchronized between vector instances
+- Pre/post processing of data - The ability to prepare and normalize data.
+- Connecting pipelines together - The ability to take input from another pipeline.
+- Component reuse - The ability to define boilerplate for reuse across many different pipelines. This will likely align with Datadog’s “pipeline catalogue”.
+- Pipeline quotas - The ability to limit how much data a pipeline can send to a sink.
+
+
+## Motivation
+
+- Helps Vector to grow organically within an organization by allowing teams to adopt Vector at their own pace without heavy administrator involvement.
+- Reduces the management overhead of devops/SREs by enabling teams to manage their own pipelines (spread the management load).
+
+
+## Internal Proposal
+
+### How and where the pipelines are stored
+
+To keep the actual structure and avoid retro compatibility issues, vector should read from the default folder `/etc/vector/pipelines/*` the difference pipeline's configuration files.
```

Comment:
I'm not totally sure what this sentence is saying. Are you stating that `/etc/vector/pipelines` will be the configuration directory for each pipeline? So, pipeline 'foo' would be configured by `/etc/vector/pipelines/foo.toml` and 'bar' `/etc/vector/pipelines/bar`?

Reply:
Yes, the pipeline foo would be configured by `/etc/vector/pipelines/foo.toml` and bar by `/etc/vector/pipelines/bar.toml` (or `json`, or `yml`). I'll clarify this.